coordinator/core/src/main/kotlin/net/consensys/zkevm/ethereum/coordination/proofcreation/ZkProofCreationCoordinatorImpl.kt

```diff
@@ -1,19 +1,41 @@
 package net.consensys.zkevm.ethereum.coordination.proofcreation
 
+import net.consensys.encodeHex
+import net.consensys.linea.async.toSafeFuture
+import net.consensys.toBigInteger
 import net.consensys.zkevm.coordinator.clients.BatchExecutionProofRequestV1
 import net.consensys.zkevm.coordinator.clients.ExecutionProverClientV2
+import net.consensys.zkevm.coordinator.clients.L2MessageServiceLogsClient
 import net.consensys.zkevm.domain.Batch
 import net.consensys.zkevm.domain.BlocksConflation
+import net.consensys.zkevm.domain.RlpBridgeLogsData
+import net.consensys.zkevm.encoding.BlockEncoder
 import net.consensys.zkevm.ethereum.coordination.conflation.BlocksTracesConflated
 import org.apache.logging.log4j.LogManager
 import org.apache.logging.log4j.Logger
+import org.web3j.protocol.Web3j
+import org.web3j.protocol.core.DefaultBlockParameter
 import tech.pegasys.teku.infrastructure.async.SafeFuture
 
 class ZkProofCreationCoordinatorImpl(
-  private val executionProverClient: ExecutionProverClientV2
+  private val executionProverClient: ExecutionProverClientV2,
+  private val l2MessageServiceLogsClient: L2MessageServiceLogsClient,
+  private val l2Web3jClient: Web3j,
+  private val encoder: BlockEncoder
```

Comment:
Move the responsibility of retrieving "L2 state root hash and bridge logs" from execution prover client to the execution proof coordinator.

Reply:
I think it makes more sense. Thanks for the change.